### PR TITLE
Numpy large tensor test batch 4

### PR DESCRIPTION
**Pull request by Zha0q1**
This is the 4th batch of the numpy large tensors tests
@josephevans @access2rohit

### CODE REVIEW

**Comment by Zha0q1** [comment]
@access2rohit  can you review?

**Review by josephevans — APPROVED** [review]

File: tests/nightly/test_np_large_array.py [hunk 1]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+
+@use_np
+def test_mean():
+    inp = np.arange(DOUBLE_INT_OVERFLOW).reshape((2, INT_OVERFLOW))
+    inp.attach_grad()
+    with mx.autograd.record():
+        out = np.mean(inp, axis=1)
+        out.backward()
+    assert out.shape == (2, )
+    assert_almost_equal(out[0], np.array((HALF_INT_OVERFLOW-0.5)), \
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:601]: can you specify dtype in the test ?

File: tests/nightly/test_np_large_array.py [hunk 2]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    with mx.autograd.record():
+        out = np.where(inp1==0, inp1, inp2)
+        out.backward()
+    assert out.shape == inp1.shape
+    assert out[0, 0] == 0 and out[-1, -1] == 2
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[0, 0] == 1 and inp1.grad[-1 ,-1] == 0
+    assert inp2.grad.shape == inp2.shape
+    assert inp2.grad[0, 0] == 0 and inp2.grad[-1 ,-1] == 1
+    # onse side is scalar
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:658]: ??

File: tests/nightly/test_np_large_array.py [hunk 3]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    with mx.autograd.record():
+        out = np.where(inp1==0, inp1, 2)
+        out.backward()
+    assert out.shape == inp1.shape
+    assert out[0, 0] == 0 and out[-1, -1] == 2
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[0, 0] == 1 and inp1.grad[-1 ,-1] == 0
+    # both sides ar scalar
+    with mx.autograd.record():
+        out = np.where(inp1==0, 0, 2)
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:668]: can we change it to `np.where(inp1==0, inp2, 2)`

File: tests/nightly/test_np_large_array.py (hunk 2 above)
Comment (Zha0q1) [reply tests/nightly/test_np_large_array.py:658]: Yeah.. I meant "one"

File: tests/nightly/test_np_large_array.py (hunk 3 above)
Comment (Zha0q1) [reply tests/nightly/test_np_large_array.py:668]: Sure

File: tests/nightly/test_np_large_array.py (hunk 1 above)
Comment (Zha0q1) [reply tests/nightly/test_np_large_array.py:601]: do you mean specify the dtype to be int?

Comment (Zha0q1) [reply tests/nightly/test_np_large_array.py:601]: here the output is expected to be somenumber.5

Comment (access2rohit) [reply tests/nightly/test_np_large_array.py:601]: I think you have improved the test now. Looks good

File: tests/nightly/test_np_large_array.py [hunk 4]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    inp2 = inp1 + 1
+    inp1.attach_grad()
+    inp2.attach_grad()
+    with mx.autograd.record():
+        out = np.where(inp1==0, inp1, inp2)
+        out.backward()
+    assert out.shape == inp1.shape
+    assert out[0, 0] == 0 and out[-1, -1] == 2
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[0, 0] == 1 and inp1.grad[-1 ,-1] == 0
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:655]: nit: comma [-1 ,-1] -> [-1, -1]

File: tests/nightly/test_np_large_array.py [hunk 5]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    assert inp2.grad.shape == inp2.shape
+    assert inp2.grad[0, 0] == 0 and inp2.grad[-1 ,-1] == 1
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:657]: same

File: tests/nightly/test_np_large_array.py [hunk 6]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    # one side is scalar
+    with mx.autograd.record():
+        out = np.where(inp1==0, inp1, 2)
+        out.backward()
+    assert out.shape == inp1.shape
+    assert out[0, 0] == 0 and out[-1, -1] == 2
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[0, 0] == 1 and inp1.grad[-1 ,-1] == 0
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:665]: same

File: tests/nightly/test_np_large_array.py [hunk 7]
```diff
@@ -542,6 +564,113 @@ def test_slice_assign():
...
+    # both sides ar scalar
+    with mx.autograd.record():
+        out = np.where(inp1==0, 0, 2)
+        out.backward()
+    assert out.shape == inp1.shape
+    assert out[0, 0] == 0 and out[-1, -1] == 2
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[-1 ,-1] == 0
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:673]: same

**Review by access2rohit — COMMENTED** [review]
Last nitpicky comments. Rest LGTM :)

**Review by access2rohit — APPROVED** [review]

**Pull request merged**